### DISCUSSION TITLE
Laziness support (or not)

### DISCUSSION THREAD

**Discussion by zhming0** [jank-lang/jank#44] [Issue Triage] [answered]
Hey, just curious if you've considered adding lazy evaluation to the project? Personally, I reckon Clojure's approach to laziness causes more harm than good. It's prone to errors, hampers performance, and quickly complicates the compiler and core library. The benefits can be achieved via 3rd party libraries instead. What's your take on it?

**Comment by jeaye** [comment, marked as answer]
Hey! jank is, first and foremost, a Clojure dialect. This means it will the primary goal is to make it behave just like Clojure, but with a native host with C++ and LLVM.

Once that's accomplished, there's room for exploration, regarding the type system, evaluation mechanism, etc, but all of that comes later. From my experience with Clojure, lazy vs. eager evaluation hasn't been high on my list of gripes.

**Reply by dmg46664** [reply]
I think that Clojure strikes a very good balance o Laziness and pragmatism. I. default `chunking behaviour https://ask.clojure.org/index.php/11687/is-lazy-sequence-chunking-deterministic

If there are case examples that could persuade all Clojurians to set sail on a different destination, that's fair enough, but I'd prefer `jank` sticks to the stanard so far, short of very compelling elucidated arguments to the counter.

**Comment by amano-kenji** [comment]
I didn't have a problem with laziness in haskell and clojure, but I never wrote a performance-sensitive application.

**Reply by jeaye** [reply]
Haskell is a lazy language. Clojure is an eager language with *some* lazy sequences. Laziness is entirely avoidable in Clojure, if one desires. That's not the case with Haskell.
